diff --git a/node_modules/react-native-screens/android/src/main/jni/CMakeLists.txt b/node_modules/react-native-screens/android/src/main/jni/CMakeLists.txt
index ec34a20..ef088f9 100644
--- a/node_modules/react-native-screens/android/src/main/jni/CMakeLists.txt
+++ b/node_modules/react-native-screens/android/src/main/jni/CMakeLists.txt
@@ -12,7 +12,7 @@ set(LIB_ANDROID_GENERATED_COMPONENTS_DIR ${LIB_ANDROID_GENERATED_JNI_DIR}/react/
 add_compile_options(
   -fexceptions
   -frtti
-  -std=c++20
+  -std=c++17
   -Wall
   -Wpedantic
   -Wno-gnu-zero-variadic-macro-arguments
@@ -50,8 +50,6 @@ target_link_libraries(
   react_render_debug
   react_render_graphics
   react_render_mapbuffer
-  react_render_componentregistry
-  react_utils
   rrc_view
   turbomodulejsijni
   yoga
@@ -63,7 +61,7 @@ target_compile_options(
   -DLOG_TAG=\"ReactNative\"
   -fexceptions
   -frtti
-  -std=c++20
+  -std=c++17
   -Wall
 )
 
diff --git a/node_modules/react-native-screens/common/cpp/react/renderer/components/rnscreens/RNSModalScreenComponentDescriptor.h b/node_modules/react-native-screens/common/cpp/react/renderer/components/rnscreens/RNSModalScreenComponentDescriptor.h
index 1c6fa19..95429db 100644
--- a/node_modules/react-native-screens/common/cpp/react/renderer/components/rnscreens/RNSModalScreenComponentDescriptor.h
+++ b/node_modules/react-native-screens/common/cpp/react/renderer/components/rnscreens/RNSModalScreenComponentDescriptor.h
@@ -12,29 +12,29 @@ class RNSModalScreenComponentDescriptor final
  public:
   using ConcreteComponentDescriptor::ConcreteComponentDescriptor;
 
-  void adopt(ShadowNode& shadowNode) const override {
-    react_native_assert(
-        dynamic_cast<RNSModalScreenShadowNode*>(&shadowNode));
-    auto& screenShadowNode =
-        static_cast<RNSModalScreenShadowNode&>(shadowNode);
-
-    react_native_assert(
-        dynamic_cast<YogaLayoutableShadowNode*>(&screenShadowNode));
-    auto& layoutableShadowNode =
-        dynamic_cast<YogaLayoutableShadowNode&>(screenShadowNode);
-
-    auto state =
-        std::static_pointer_cast<const RNSModalScreenShadowNode::ConcreteState>(
-            shadowNode.getState());
-    auto stateData = state->getData();
-
-    if (stateData.frameSize.width != 0 && stateData.frameSize.height != 0) {
-      layoutableShadowNode.setSize(
-          Size{stateData.frameSize.width, stateData.frameSize.height});
-    }
-
-    ConcreteComponentDescriptor::adopt(shadowNode);
-  }
+        void adopt(ShadowNode::Unshared const &shadowNode) const override {
+          react_native_assert(
+              std::dynamic_pointer_cast<RNSScreenShadowNode>(shadowNode));
+          auto screenShadowNode =
+              std::static_pointer_cast<RNSModalScreenShadowNode>(shadowNode);
+
+          react_native_assert(
+              std::dynamic_pointer_cast<YogaLayoutableShadowNode>(screenShadowNode));
+          auto layoutableShadowNode =
+              std::static_pointer_cast<YogaLayoutableShadowNode>(screenShadowNode);
+
+          auto state =
+              std::static_pointer_cast<const RNSModalScreenShadowNode::ConcreteState>(
+                  shadowNode->getState());
+          auto stateData = state->getData();
+
+          if (stateData.frameSize.width != 0 && stateData.frameSize.height != 0) {
+            layoutableShadowNode->setSize(
+                Size{stateData.frameSize.width, stateData.frameSize.height});
+          }
+
+          ConcreteComponentDescriptor::adopt(shadowNode);
+        }
 };
 
 } // namespace react
diff --git a/node_modules/react-native-screens/common/cpp/react/renderer/components/rnscreens/RNSScreenComponentDescriptor.h b/node_modules/react-native-screens/common/cpp/react/renderer/components/rnscreens/RNSScreenComponentDescriptor.h
index 67194d3..40a3c0a 100644
--- a/node_modules/react-native-screens/common/cpp/react/renderer/components/rnscreens/RNSScreenComponentDescriptor.h
+++ b/node_modules/react-native-screens/common/cpp/react/renderer/components/rnscreens/RNSScreenComponentDescriptor.h
@@ -12,24 +12,24 @@ class RNSScreenComponentDescriptor final
  public:
   using ConcreteComponentDescriptor::ConcreteComponentDescriptor;
 
-  void adopt(ShadowNode& shadowNode) const override {
+  void adopt(ShadowNode::Unshared const &shadowNode) const override {
     react_native_assert(
-        dynamic_cast<RNSScreenShadowNode*>(&shadowNode));
-    auto& screenShadowNode =
-        static_cast<RNSScreenShadowNode&>(shadowNode);
+        std::dynamic_pointer_cast<RNSScreenShadowNode>(shadowNode));
+    auto screenShadowNode =
+        std::static_pointer_cast<RNSScreenShadowNode>(shadowNode);
 
     react_native_assert(
-        dynamic_cast<YogaLayoutableShadowNode*>(&screenShadowNode));
-    auto& layoutableShadowNode =
-        dynamic_cast<YogaLayoutableShadowNode&>(screenShadowNode);
+        std::dynamic_pointer_cast<YogaLayoutableShadowNode>(screenShadowNode));
+    auto layoutableShadowNode =
+        std::static_pointer_cast<YogaLayoutableShadowNode>(screenShadowNode);
 
     auto state =
         std::static_pointer_cast<const RNSScreenShadowNode::ConcreteState>(
-            shadowNode.getState());
+            shadowNode->getState());
     auto stateData = state->getData();
 
     if (stateData.frameSize.width != 0 && stateData.frameSize.height != 0) {
-      layoutableShadowNode.setSize(
+      layoutableShadowNode->setSize(
           Size{stateData.frameSize.width, stateData.frameSize.height});
     }
 
